src/Features/CSharp/Portable/UseExpressionBody/UseExpressionBodyCodeFixProvider.cs

```diff
@@ -31,7 +31,7 @@ public UseExpressionBodyCodeFixProvider()
         }
 
         protected override bool IncludeDiagnosticDuringFixAll(Diagnostic diagnostic)
-            => diagnostic.Severity != DiagnosticSeverity.Hidden ||
```

Comment:
interesting? why explicitly filtering out hidden diagnostic for fix all? anyway, I am not sure how you get suppressed diagnostic for fix all. assuming there is no hidden IDE only hack, the engine should never give suppressed diagnostics to fix all.

Reply:
> why explicitly filtering out hidden diagnostic for fix all? 

In some projects you may have this things set to a higher level (and would get things like suggestions ellipses in the code).  In other projects, the feature level may be 'true:none' (since you want to generate expression-bodies, but you don't want to see anything in the UI.  We use this so that you can still actually generate the expression body in that project if you're on the item.

However, if you fix all, we don't want to update the nodes we marked with the hidden diagnostic.  Only teh ones the user can visibly see marked and want updated.

Reply:
so should be pushed to engine? @mavasani @jinujoseph @kuhlenh

Reply:
I think we are now seeing suppressed diagnostic instances in the IDE now for rules which are completely disabled (turned off in ruleset or tools option) after https://github.com/dotnet/roslyn/pull/27220. I mentioned this in https://github.com/dotnet/roslyn/pull/27220#pullrequestreview-125710982, and I believe we should revert that approach/workaround for a better design as soon as possible, as that violates lot of basic assumptions in the IDE diagnostic engine.

Reply:
just looked. so another case where IDE does its own then. so now IDE analyzer can bypass all engine and set severity whatever they want.. these analyzer will not be able to run in other host but only in IDE.

Reply:
Yes, in order for us to move these analyzers to NuGet, we will need to remove all performance/functional workarounds in the analyzers and/or the IDE diagnostic engine. Sam already has an idea on how to tackle dynamic diagnostic IDs for naming style analyzers, but we will need to explicitly identify and fix remaining issues like this before that work can be done.

Reply:
@mavasani 

> Sam already has an idea on how to tackle dynamic diagnostic IDs for naming style analyzers, but we will need to explicitly identify and fix remaining issues like this before that work can be done.

I would def like to be involved/aware of those issues, and how we would like ot design support for that into the analysis subsystem.  It would def be helpful.  Thanks!

Reply:
wondering how all these work with suppression service.. have we tested that area? or are we treat diagnostics from IDE analyzer special in suppression service as well? suppression service is supposed to be able to unsupress suppressed diagnostics, isn't it? does that work with IDE analyzer?
…

Reply:
> wondering how all these work with suppression service.. have we tested that area? or are we treat diagnostics from IDE analyzer special in suppression service as well? suppression service is supposed to be able to unsupress suppressed diagnostics, isn't it? does that work with IDE analyzer?

I did bring up these points on the PR from @sharwell. Sam were you able to test if suppressing diagnostic from error list, then unsuppressing source-suppressed diagnostics work fine? We might have just broken that error list feature.

Reply:
> Sam were you able to test if suppressing diagnostic from error list, then unsuppressing source-suppressed diagnostics work fine? We might have just broken that error list feature.

Not yet. I'm trying to round out the more obvious items and looking at ones like this for preview 5 polish.

Reply:
> In other projects, the feature level may be 'true:none'

`true:none` in .editorconfig now acts as a suppression (or disabled), so these would not appear in the Fix All experience. `silent` severity *would* now show in the Fix All experience.

Reply:
> suppression service is supposed to be able to unsupress suppressed diagnostics, isn't it?

👍 This is the part I was missing. I knew about the ability to filter the error list, but that alone seemed like a really small issue.

Reply:
> Sam were you able to test if suppressing diagnostic from error list, then unsuppressing source-suppressed diagnostics work fine? We might have just broken that error list feature.

I tested this today, and it appears the use of `none` in .editorconfig prevents the items from showing up in Error List even if the list isn't filtered to Active.